[BUG] thread annotation problem for mutex when using libc++ and enabled libc++'s own thread annotation

**Describe the bug**
We are using clang 8 with libc++ with libc++'s thread annotation enabled by `-D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS` globally.  However, it conflicts with benchmark's mutex wrapper's thread annotation.

**System**
Which OS, compiler, and compiler version are you using:
  - OS: Linux
  - Compiler and version:  clang 8 with libc++

**To reproduce**
Steps to reproduce the behavior:
1. sync to commit 8e0b1913d4ea803dfeb2e55567208fcab6b1b6c7
2. 
```
export CC=clang 
bazel build --copt '-stdlib=libc++' --copt -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS :benchmark
```
4. See error
```
src/mutex.h:79:40: warning: mutex 'mut_' is still held at the end of function [-Wthread-safety-analysis]
  void lock() ACQUIRE() { mut_.lock(); }
                                       ^
src/mutex.h:79:32: note: mutex acquired here
  void lock() ACQUIRE() { mut_.lock(); }
                               ^
src/mutex.h:80:34: warning: releasing mutex 'mut_' that was not held [-Wthread-safety-analysis]
  void unlock() RELEASE() { mut_.unlock(); }
                                 ^
```
**Expected behavior**
Should compile without warning

I've encountered this as well.

I believe that unrelated changes have caused the `HAVE_THREAD_SAFETY_ATTRIBUTES` check to fail during configuration, so `-DHAVE_THREAD_SAFETY_ATTRIBUTES` isn't set and the annotations aren't actually added. If I manually add `-DHAVE_THREAD_SAFETY_ATTRIBUTES` to my `CMAKE_CXX_FLAGS`, the warnings go away.

Here's the link error in the `HAVE_THREAD_SAFETY_ATTRIBUTES` check:
```
/usr/bin/ld: CMakeFiles/cmTC_2984c.dir/thread_safety_attributes.cpp.o: in function `__cxx_global_var_init':
thread_safety_attributes.cpp:(.text.startup+0x5): undefined reference to `benchmark::internal::InitializeStreams()'
```

--- a/cmake/thread_safety_attributes.cpp
+++ b/cmake/thread_safety_attributes.cpp
@@ -1,4 +1,25 @@
-#define HAVE_THREAD_SAFETY_ATTRIBUTES
-#include "../src/mutex.h"
+#if defined(__clang__)
+#define THREAD_ANNOTATION_ATTRIBUTE_(x) __attribute__((x))
+#else
+#define THREAD_ANNOTATION_ATTRIBUTE_(x)
+#endif
 
-int main() {}
+#define CAPABILITY(x) THREAD_ANNOTATION_ATTRIBUTE_(capability(x))
+#define ACQUIRE() THREAD_ANNOTATION_ATTRIBUTE_(acquire_capability())
+#define RELEASE() THREAD_ANNOTATION_ATTRIBUTE_(release_capability())
+
+class CAPABILITY("mutex") Mutex {
+ public:
+  void lock() ACQUIRE();
+  void unlock() RELEASE();
+};
+
+void Mutex::lock() ACQUIRE() {}
+void Mutex::unlock() RELEASE() {}
+
+int main() {
+  Mutex m;
+  m.lock();
+  m.unlock();
+  return 0;
+}

